### PR TITLE
Mitigate impact of #342

### PR DESCRIPTION
**Pull request by bwohlberg**
Mitigate impact of #342.

### CODE REVIEW

File: scico/denoiser.py [hunk 1]
```diff
@@ -5,7 +5,26 @@
 # user license can be found in the 'LICENSE' file distributed with the
 # package.
 
-"""Denoisers."""
+"""
```
Comment (tbalke) [inline scico/denoiser.py:8]: The user sees this when looking carefully at the source code or with `help(scico.denoiser)`. Do you think this is enough? Perhaps if these packages are installed there could be an additional warning at the time of import?

**Review by tbalke — APPROVED** [review]

File: scico/denoiser.py (hunk 1 above)
Comment (bwohlberg) [reply scico/denoiser.py:8]: The user should also see this when looking at the online API docs for `scico.denoisers`. Good question as to whether it's enough. A warning on importing the module is not a bad idea.

File: scico/denoiser.py [hunk 2]
```diff
@@ -8,22 +8,18 @@
...
-numerical precision on all other calculations run within the same
-process, even if neither of :func:`bm3d` or :func:`bm4d` are actually
-used. Users who are not making use of either :func:`bm3d` or
-:func:`bm4d` are advised not to install the corresponding packages.
+for the duration of the process that loads it. If this package is
+installed, simply loading this module (:mod:`scico.denoiser`), is
+sufficient to inflict this loss of numerical precision on all other
+calculations run within the same process, even if :func:`bm4d` is not
+actually used. Users who are not making use of :func:`bm4d` are
+advised not to install the corresponding package. For additional information, see `scico issue #342 <https://github.com/lanl/scico/issues/342>`__.
```
Comment (tbalke) [inline scico/denoiser.py:22]: line too long

Comment (bwohlberg) [reply scico/denoiser.py]: Fixed.

File: scico/denoiser.py [hunk 3]
```diff
@@ -25,6 +41,19 @@
...
+    if hasattr(finfo, "smallest_subnormal"):  # can't test with older numpy versions
+        if finfo.smallest_subnormal == 0.0:
+            warnings.warn(
+                "Importing module bm4d has had an adverse affect on floating point "
+                "accuracy. See the documentation for module scico.denoiser, and "
+                "scico issue #342."
+            )
+            warnings.filterwarnings(
+                action="ignore",
+                message="^The value of the smallest subnormal",
```
Comment (tbalke) [inline scico/denoiser.py:56]: is this caret character there on purpose?

**Review by tbalke — APPROVED** [review]

File: scico/denoiser.py (hunk 3 above)
Comment (bwohlberg) [reply scico/denoiser.py:56]: Yes, `message` is a regex.

**Pull request merged**